diff -rupN gopacket/layers/enums.go gopacket_patched/layers/enums.go
--- gopacket/layers/enums.go	2015-02-26 14:25:12.000000000 +0100
+++ gopacket_patched/layers/enums.go	2015-02-26 14:41:50.694724469 +0100
@@ -96,6 +96,7 @@ const (
 	LinkTypeSLIP           LinkType = 8
 	LinkTypePPP            LinkType = 9
 	LinkTypeFDDI           LinkType = 10
+	LinkTypeOVPN           LinkType = 12 // OSAG fix to support OpenVPN tunnels
 	LinkTypeATM_RFC1483    LinkType = 100
 	LinkTypeRaw            LinkType = 101
 	LinkTypePPP_HDLC       LinkType = 50
@@ -480,6 +481,10 @@ func init() {
 	LinkTypeMetadata[LinkTypeNull] = EnumMetadata{DecodeWith: gopacket.DecodeFunc(decodeLoopback), Name: "Null"}
 	LinkTypeMetadata[LinkTypeLoop] = EnumMetadata{DecodeWith: gopacket.DecodeFunc(decodeLoopback), Name: "Loop"}
 	LinkTypeMetadata[LinkTypeRaw] = EnumMetadata{DecodeWith: gopacket.DecodeFunc(decodeIPv4or6), Name: "Raw"}
+
+	// link type decoder settings added by OSAG
+    LinkTypeMetadata[LinkTypeOVPN] = EnumMetadata{DecodeWith: gopacket.DecodeFunc(decodeIPv4or6), Name: "OVPN"}
+
 	LinkTypeMetadata[LinkTypePFLog] = EnumMetadata{DecodeWith: gopacket.DecodeFunc(decodePFLog), Name: "PFLog"}
 	LinkTypeMetadata[LinkTypeIEEE80211Radio] = EnumMetadata{DecodeWith: gopacket.DecodeFunc(decodeRadioTap), Name: "RadioTap"}
 	LinkTypeMetadata[LinkTypeLinuxUSB] = EnumMetadata{DecodeWith: gopacket.DecodeFunc(decodeUSB), Name: "USB"}
diff -rupN gopacket/packet.go gopacket_patched/packet.go
--- gopacket/packet.go	2015-02-26 14:25:12.000000000 +0100
+++ gopacket_patched/packet.go	2015-02-26 14:42:31.958265088 +0100
@@ -28,6 +28,9 @@ type CaptureInfo struct {
 	// Length is the size of the original packet.  Should always be >=
 	// CaptureLength.
 	Length int
+	// OSAG addon: inbound specifies the direction from which the interface
+	// receives the packet
+	Inbound uint8
 }
 
 // PacketMetadata contains metadata for a packet.
diff -rupN gopacket/pcap/pcap.go gopacket_patched/pcap/pcap.go
--- gopacket/pcap/pcap.go	2014-11-12 00:43:59.000000000 +0100
+++ gopacket_patched/pcap/pcap.go	2015-02-26 14:41:00.483275182 +0100
@@ -316,6 +316,10 @@ func (p *Handle) getNextBufPtrLocked(ci
 		int64(p.pkthdr.ts.tv_usec)*1000) // convert micros to nanos
 	ci.CaptureLength = int(p.pkthdr.caplen)
 	ci.Length = int(p.pkthdr.len)
+
+	// OSAG addon: add the direction to the metadata
+	ci.Inbound = uint8(p.pkthdr.inbound)
+
 	return nil
 }
 
