diff -rupN libpcap-1.5.3/pcap/pcap.h libpcap-1.5.3_mod/pcap/pcap.h
--- libpcap-1.5.3/pcap/pcap.h	2013-11-08 00:23:22.000000000 +0100
+++ libpcap-1.5.3_mod/pcap/pcap.h	2014-12-23 10:03:21.634016353 +0100
@@ -163,6 +163,7 @@ struct pcap_pkthdr {
 	struct timeval ts;	/* time stamp */
 	bpf_u_int32 caplen;	/* length of portion present */
 	bpf_u_int32 len;	/* length this packet (off wire) */
+    u_int8_t inbound;
 };
 
 /*
diff -rupN libpcap-1.5.3/pcap-linux.c libpcap-1.5.3_mod/pcap-linux.c
--- libpcap-1.5.3/pcap-linux.c	2014-01-14 21:40:35.000000000 +0100
+++ libpcap-1.5.3_mod/pcap-linux.c	2014-12-23 10:03:36.393844549 +0100
@@ -4211,6 +4211,13 @@ static int pcap_handle_packet_mmap(
 		return 0;
 
 	sll = (void *)frame + TPACKET_ALIGN(handlep->tp_hdrlen);
+
+    /* OSAG: set packet direction */
+    if (sll->sll_pkttype == PACKET_OUTGOING)
+        pcaphdr.inbound = 0;
+    else
+        pcaphdr.inbound = 1;
+
 	if (!linux_check_direction(handle, sll))
 		return 0;
 
